packages/mermaid/src/diagrams/sequence/sequenceDb.ts

```diff
@@ -516,10 +513,10 @@ export const apply = function (param: any | AddMessageParams | AddMessageParams[
         state.records.destroyedActors[param.actor] = state.records.messages.length;
         break;
       case 'activeStart':
-        addSignal(param.actor, undefined, undefined, param.signalType);
+        addSignal(param.actor.actor, undefined, undefined, param.signalType);
         break;
       case 'activeEnd':
-        addSignal(param.actor, undefined, undefined, param.signalType);
+        addSignal(param.actor.actor, undefined, undefined, param.signalType);
         break;
       case 'addNote':
         addNote(param.actor, param.placement, param.text);
```

Comment:
Why is a `.actor` being added here, while it's removed everywhere else?

It also deviates from the uniform way where `param.actor` is the first parameter in other functions. 
 
Maybe we should modify the addSignal function to accept the right type data?

Reply:
The reason is in this case the input is 
```
{actor: { actor: 'John' } }
```

Hence for these two cases `activeStart` and `activeEnd` I am passing `param.actor.actor` as `addSignal` except s a string param and not object anymore.

Reply:
Was it not working previously then?
Where is the input coming from? Can we fix it there?

Reply:
yes, it was working previously since the actor was an object. I tried looking deeper into this earlier to fix it the caller instead but wasn't sure where exactly I should be fixing it.
The syntax of activation / deactivation leads to the actor being created as an object

```
Alice->>+John: Yes
John-->>-Alice: OK
```

Could you point me to the right direction of where I can fix the above to generate actor as a string instead of object ?

Reply:
@sidharthv96 did you get a chance to check this ?

Reply:
@ad1992 I was able to fix by tweaking the jison. I'll push it, let's see if any tests fail. We can revert if it does.